### PR TITLE
Improve evm.memory_access test

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

**Comment by gumb0** [comment]
@chfast 2Gb + 2Gb case fails currently for evmone, too

**Comment by gumb0** [comment]
Passes with 30M gas

File: test/unittests/evm_test.cpp [hunk 1]
```diff
@@ -1171,16 +1173,24 @@ TEST_F(evm, memory_access)
 
             code += bytecode{t.opcode};
 
-            execute(code);
+            auto const gas = 30000000;
```
Comment (chfast) [inline test/unittests/evm_test.cpp]: Can you try `8796294610952`. That's the max value causing all test to go OOG

Comment (gumb0) [reply test/unittests/evm_test.cpp]: evmone passes, interpreter returns `EVMC_INVALID_MEMORY_ACCESS`

Comment (chfast) [reply test/unittests/evm_test.cpp]: Is this solved now?

Comment (gumb0) [reply test/unittests/evm_test.cpp]: No, interpreter checks the size of returndata first and returns `EVMC_INVALID_MEMORY_ACCESS` before checking for required memory size

Comment (chfast) [reply test/unittests/evm_test.cpp]: Can we apply something like this then? https://github.com/ethereum/evmone/pull/92

It also looks `RETURNDATACOPY` might require its own test for `OUT_OF_GAS` not being `INVALID_MEMORY_ACCESS`.

Comment (gumb0) [reply test/unittests/evm_test.cpp]: Yes that, helps.

I rebased this PR on #92

Comment (chfast) [reply test/unittests/evm_test.cpp]: Ready now?

Comment (gumb0) [reply test/unittests/evm_test.cpp]: Yes

**Pull request merged**